### PR TITLE
Improved OVAL and OCIL generator elements

### PR DESCRIPTION
**Pull request by mpreisler**
Now we list which script generator it and which version of SSG. Previously we just include "python" and version of python.

New OVAL generator element:
```
  <ns0:generator>
    <ns2:product_name>combine-ovals.py from SCAP Security Guide</ns2:product_name>
    <ns2:product_version>ssg: 0.1.34, python: 2.7.13</ns2:product_version>
    <ns2:schema_version>5.11</ns2:schema_version>
    <ns2:timestamp>2017-07-10T19:38:34</ns2:timestamp>
  </ns0:generator>
```

New OCIL generator element:
```
  <ns0:generator>
    <ns0:product_name>xccdf-create-ocil.xslt from SCAP Security Guide</ns0:product_name>
    <ns0:product_version>ssg: 0.1.34</ns0:product_version>
    <ns0:schema_version>2.0</ns0:schema_version>
    <ns0:timestamp>2017-07-10T15:42:53-04:00</ns0:timestamp>
  </ns0:generator>
```

I also added support for detecting OpenSCAP version at configure time. I thought I would add it to the generator elements but to be fair `oscap` is not used for OVAL and OCIL. Still, I think it will be handy in the future and it just says "unknown" if the regex fails so I kept it in the PR.

### CODE REVIEW

File: shared/transforms/xccdf-create-ocil.xslt [hunk 1]
```diff
@@ -5,15 +5,19 @@ xmlns:cdf="http://checklists.nist.gov/xccdf/1.1" exclude-result-prefixes="cdf"
...
 
   <xsl:template match="/">
   <ocil xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://scap.nist.gov/schema/ocil/2.0" >
    <generator>
-   <schema_version>2.0</schema_version>
-   <timestamp><xsl:value-of as="xs:dateTime" select="date:date-time()"/></timestamp>
+     <product_name>xccdf-create-ocil.xslt from SCAP Security Guide</product_name>
+     <product_version>ssg: <xsl:value-of select="$ssg_version"/></product_version>
+     <schema_version>2.0</schema_version>
+     <timestamp><xsl:value-of as="xs:dateTime" select="date:date-time()"/></timestamp>
```
Comment (dahaic) [inline shared/transforms/xccdf-create-ocil.xslt]: This produces a bit different format of timestamp to the rest of components (being timezone adjusted). Is it fixable?

Comment (mpreisler) [inline shared/transforms/xccdf-create-ocil.xslt]: I looked into it in the past and UTC date time is pretty difficult to pull off in XSLT 1.0. They fixed it in XSLT 2.0 but we can't use that for SSG.

Comment (mpreisler) [inline shared/transforms/xccdf-create-ocil.xslt]: I could make it timezone adjusted in the OVAL doc.

File: shared/transforms/xccdf-create-ocil.xslt [hunk 2]
```diff
@@ -5,15 +5,19 @@ xmlns:cdf="http://checklists.nist.gov/xccdf/1.1" exclude-result-prefixes="cdf"
...
 <!-- This transform expects checks with system "ocil-transitional" and that these contain check-content
      that can transformed into OCIL questionnaires.
      -->
 
   <xsl:template match="/">
   <ocil xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://scap.nist.gov/schema/ocil/2.0" >
    <generator>
-   <schema_version>2.0</schema_version>
-   <timestamp><xsl:value-of as="xs:dateTime" select="date:date-time()"/></timestamp>
+     <product_name>xccdf-create-ocil.xslt from SCAP Security Guide</product_name>
```
Comment (redhatrises) [inline shared/transforms/xccdf-create-ocil.xslt]: I think that this should only be `SCAP Security Guide`

File: shared/utils/combine-ovals.py [hunk 3]
```diff
@@ -54,11 +48,12 @@ def _header(schema_version):
         http://oval.mitre.org/XMLSchema/oval-definitions-5#unix unix-definitions-schema.xsd
         http://oval.mitre.org/XMLSchema/oval-definitions-5#linux linux-definitions-schema.xsd">
     <generator>
-        <oval:product_name>python</oval:product_name>
-        <oval:product_version>%s</oval:product_version>
+        <oval:product_name>combine-ovals.py from SCAP Security Guide</oval:product_name>
```
Comment (redhatrises) [inline shared/utils/combine-ovals.py]: I think that this should only be `SCAP Security Guide`

File: shared/utils/combine-ovals.py [hunk 4]
```diff
@@ -54,11 +48,12 @@ def _header(schema_version):
...
+        <oval:product_version>ssg: %s, python: %s</oval:product_version>
```
Comment (redhatrises) [inline shared/utils/combine-ovals.py]: What is the point of having python in the product version? Shouldn't this only be the SSG product version?

File: shared/transforms/xccdf-create-ocil.xslt [hunk 5]
```diff
@@ -5,15 +5,19 @@ xmlns:cdf="http://checklists.nist.gov/xccdf/1.1" exclude-result-prefixes="cdf"
...
+     <product_version>ssg: <xsl:value-of select="$ssg_version"/></product_version>
```
Comment (redhatrises) [inline shared/transforms/xccdf-create-ocil.xslt]: I would just make this `<xsl:value-of select="$ssg_version"/>`

File: shared/transforms/xccdf-create-ocil.xslt (hunk 1 above)
Comment (dahaic) [inline shared/transforms/xccdf-create-ocil.xslt]: I vote for consistency across the components :)

File: shared/utils/combine-ovals.py (hunk 4 above)
Comment (mpreisler) [inline shared/utils/combine-ovals.py]: it helps with debugging

File: shared/utils/combine-ovals.py (hunk 3 above)
Comment (mpreisler) [inline shared/utils/combine-ovals.py]: it helps with debugging

**Comment by mpreisler** [comment]
I think having the script in generator helps with debugging. I am willing to remove it if there is consensus though.

File: shared/utils/combine-ovals.py (hunk 4 above)
Comment (redhatrises) [inline shared/utils/combine-ovals.py]: End product wise.... It doesn't make much sense.

File: shared/utils/combine-ovals.py (hunk 3 above)
Comment (redhatrises) [inline shared/utils/combine-ovals.py]: End product wise.... It doesn't have sense.

**Comment by redhatrises** [comment]
True. But in the end, it isn't accurate as the end product name is "SCAP Security Guide" and not "combine-ovals.py"

**Comment by dahaic** [comment]
From my point of view it is valid to have a script name there, as the script is the generator of particular component.

File: shared/utils/combine-ovals.py (hunk 3 above)
Comment (mpreisler) [inline shared/utils/combine-ovals.py]: :D IMO SCAP Security Guide is the product, combine-ovals.py is the generator. it makes perfect sense to me. It's not supposed to list the product, it's supposed to list the generator product. Hope that makes sense.

**Comment by mpreisler** [comment]
> True. But in the end, it isn't accurate as the end product name is "SCAP Security Guide" and not "combine-ovals.py"

`generator/product_name` is not supposed to be the name of the generated product. It's the product name of the generator that generated the product.

**Comment by redhatrises** [comment]
/me

image

After looking at the OVAL specification, ignore all my stupid and wrong comments. Apparently, brain is off kilter on this Monday.... see #openscap irc comments from earlier today..... *sigh*

**Comment by redhatrises** [comment]
@mpreisler are you going to work on anything TZ related to this PR?

**Comment by mpreisler** [comment]
> After looking at the OVAL specification, ignore all my stupid and wrong comments. Apparently, brain is off kilter on this Monday.... see #openscap irc comments from earlier today..... sigh

no worries, happens to the best of us.

**Comment by mpreisler** [comment]
> @mpreisler are you going to work on anything TZ related to this PR?

If somebody is willing to merge this I would love to avoid timezone shenanigans.

**Comment by redhatrises** [comment]
Ack.

**Pull request merged**